exist-core/src/main/java/org/exist/xquery/Predicate.java

```diff
@@ -194,9 +194,13 @@ public Sequence evalPredicate(final Sequence outerSequence,
                         innerSeq = inner.eval(Dependency.dependsOn(inner.getDependencies(), Dependency.CONTEXT_ITEM)
                                 ? contextSequence : null);
                     }
-                    if (innerSeq.getCardinality().isSubCardinalityOrEqualOf(Cardinality.EXACTLY_ONE)) {
+
+                    if (innerSeq.getCardinality().equals(Cardinality.EMPTY_SEQUENCE)) {
+                        result = Sequence.EMPTY_SEQUENCE;
+                    } else if (innerSeq.getCardinality().isSubCardinalityOrEqualOf(Cardinality.EXACTLY_ONE)) {
                         result = selectByPosition(outerSequence, contextSequence, mode, innerSeq);
-                    } else {
+                    }
```

Comment:
formatting is not ok?

Reply:
It depends, I would write all else blocks that way. The implicit coding style of eXist-db is different, though. I will change that.